root/programs/VB/Frameworks/Infrastructure/Business/Business/MyApsBaseLogic.vb

```diff
@@ -0,0 +1,584 @@
+
+                ' Log4Netへログ出力
+                LogIF.WarnLog("ACCESS", strLogMessage)
+            End If
+
+            ' -------------------------------------------------------------   
+        End Sub
+
+        ''' <summary>
+        ''' Ｂ層のシステム例外による異常終了の後処理を実装するUOCメソッド。
+        ''' </summary>
+        ''' <param name="parameterValue">引数クラス</param>
+        ''' <param name="returnValue">戻り値クラス</param>
+        ''' <param name="bsEx">BusinessSystemException</param>
+        ''' <remarks>業務コード親クラス１から利用される派生の末端</remarks>
+        Protected Overrides Sub UOC_ABEND(ByVal parameterValue As BaseParameterValue, ByVal returnValue As BaseReturnValue, ByVal bsEx As BusinessSystemException)
+            ' システム例外発生時の処理を実装
+            ' TODO:
+
+            ' nullチェック
+            ' なにもしない
+            If Me.perfRec Is Nothing Then
+            Else
+                ' 性能測定終了
+                Me.perfRec.EndsPerformanceRecord()
+
+                ' ACCESSログ出力-----------------------------------------------
+
+                Dim myPV As MyParameterValue = DirectCast(parameterValue, MyParameterValue)
+
+                ' ------------
+                ' メッセージ部
+                ' ------------
+                ' ユーザ名, IPアドレス, レイヤ, 
+                ' 画面名, コントロール名, メソッド名, 処理名
+                ' 処理時間（実行時間）, 処理時間（CPU時間）
+                ' エラーメッセージID, エラーメッセージ等
+                ' ------------
+                Dim strLogMessage As String = ((((("," & myPV.User.UserName & ",") & myPV.User.IPAddress & "," & "<<-----" & ",") & myPV.ScreenId & ",") & myPV.ControlId & ",") & myPV.MethodName & ",") & myPV.ActionType & "," & Convert.ToString(Me.perfRec.ExecTime) & "," & Convert.ToString(Me.perfRec.CpuTime) & "," & Convert.ToString(bsEx.messageID) & "," & Convert.ToString(bsEx.Message) & vbCr & vbLf
+                ' OriginalStackTrace（ログ出力）の品質向上
+                If Me.OriginalStackTrace = "" Then
+                    strLogMessage &= bsEx.StackTrace
+                Else
+                    strLogMessage &= Me.OriginalStackTrace
+                End If
+
+                ' Log4Netへログ出力
+                LogIF.ErrorLog("ACCESS", strLogMessage)
+            End If
+
+            ' -------------------------------------------------------------   
+        End Sub
+
+        ''' <summary>
+        ''' Ｂ層の一般的な例外による異常終了の後処理を実装するUOCメソッド。
+        ''' </summary>
+        ''' <param name="parameterValue">引数クラス</param>
+        ''' <param name="returnValue">戻り値クラス</param>
+        ''' <param name="ex">Exception</param>
+        ''' <remarks>業務コード親クラス１から利用される派生の末端</remarks>
+        Protected Overrides Sub UOC_ABEND(ByVal parameterValue As BaseParameterValue, ByRef returnValue As BaseReturnValue, ByVal ex As Exception)
+            ' 一般的な例外発生時の処理を実装
+            ' TODO:
+
+            ' nullチェック
+            If Me.perfRec Is Nothing Then
+                ' なにもしない
+
+                ' リスロー
```

Comment:
I had thought that this is problem of SharpDeveloper.
But, C# version is also the same state.

I think that it is good to move the following code to the following locations.
https://github.com/OpenTouryoProject/OpenTouryo/blob/develop/root/programs/C%23/Frameworks/Infrastructure/Business/Business/MyApsBaseLogic.cs

- Delete
```C#
572 // なにもしない 
573 
574 // リスロー 
575 throw ex; 
````
- Delete
```C#
717 // リスロー 
718 throw ex; 
````
  ↓ ↓ ↓
- Add
```C#
723 } 
000 // リスロー 
000 throw ex; 
725 #endregion 

````